components/esp_lvgl_port/include/esp_lvgl_port_knob.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief ESP LVGL port knob
+ */
+
+#pragma once
+
+#include "esp_err.h"
+#include "lvgl.h"
+
+#if __has_include ("iot_knob.h")
+#include "iot_knob.h"
+#include "iot_button.h"
```

Comment:
When I tried adding `knob` without `button` component the compilation failed here.

Knob does not depend on button in the original component. Could we remove the dependency here? (Or at least not fail in this case?)

Reply:
We need `button` component for this file: `esp_lvgl_port_knob.c`. When we add knob, there is code for handle click by `button` component.

Reply:
I see, could we do something like
``` c
#if __has_include ("iot_knob.h")
#if !__has_include("iot_button.h")
#error LVLG Knob requires button component. Please add it with idf.py add-dependency espressif/button
#endif
#include "iot_knob.h"
#include "iot_button.h"